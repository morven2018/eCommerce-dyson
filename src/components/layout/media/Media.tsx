import styles from './Media.module.scss';

export default function Media() {
  return (
    <section className={styles.media}>
      <h2 className={styles.title}>Follow us on social media</h2>
      <div className={styles.subtitle}>
        For the latest product updates and innovations!
      </div>
      <div className={styles.container}>
        <a
          href="https://x.com/dyson"
          rel="noreferrer"
          target="_blank"
          className={`${styles.link} ${styles.linkX}`}
        >
          <span className={`${styles.icon} ${styles.iconX}`}></span>
        </a>
        <a
          href="https://www.instagram.com/dysonusa/"
          rel="noreferrer"
          target="_blank"
          className={styles.link}
        >
          <span className={`${styles.icon} ${styles.iconI}`}></span>
        </a>
        <a
          href="https://www.youtube.com/dyson"
          rel="noreferrer"
          target="_blank"
          className={styles.link}
        >
          <span className={`${styles.icon} ${styles.iconY}`}></span>
        </a>
        <a
          href="https://www.tiktok.com/@dyson_usa"
          rel="noreferrer"
          target="_blank"
          className={styles.link}
        >
          <span className={`${styles.icon} ${styles.iconT}`}></span>
        </a>
      </div>
    </section>
  );
}
