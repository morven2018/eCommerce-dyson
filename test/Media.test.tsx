import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import Media from '../src/components/layout/media/Media';
import '@testing-library/jest-dom/vitest';

vi.mock('../src/components/Media.module.scss', () => ({
  default: {
    media: 'media',
    title: 'title',
    subtitle: 'subtitle',
    container: 'container',
    link: 'link',
    linkX: 'linkX',
    icon: 'icon',
    iconX: 'iconX',
    iconI: 'iconI',
    iconY: 'iconY',
    iconT: 'iconT',
  },
}));

describe('Media Component', () => {
  it('renders title and subtitle', () => {
    render(<Media />);

    expect(screen.getByText('Follow us on social media')).toBeInTheDocument();
    expect(
      screen.getByText('For the latest product updates and innovations!')
    ).toBeInTheDocument();
  });

  it('renders social media links with correct attributes and classes', () => {
    render(<Media />);

    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(4);

    expect(links[0]).toHaveAttribute('href', 'https://x.com/dyson');
    expect(links[0]).toHaveAttribute('rel', 'noreferrer');
    expect(links[0]).toHaveAttribute('target', '_blank');

    expect(links[1]).toHaveAttribute(
      'href',
      'https://www.instagram.com/dysonusa/'
    );
    expect(links[1]).toHaveAttribute('rel', 'noreferrer');
    expect(links[1]).toHaveAttribute('target', '_blank');

    expect(links[2]).toHaveAttribute('href', 'https://www.youtube.com/dyson');
    expect(links[2]).toHaveAttribute('rel', 'noreferrer');
    expect(links[2]).toHaveAttribute('target', '_blank');

    expect(links[3]).toHaveAttribute(
      'href',
      'https://www.tiktok.com/@dyson_usa'
    );
    expect(links[3]).toHaveAttribute('rel', 'noreferrer');
    expect(links[3]).toHaveAttribute('target', '_blank');
  });
});
