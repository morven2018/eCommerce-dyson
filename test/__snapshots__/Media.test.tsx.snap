// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Media Component > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="_media_ea1127"
  >
    <h2
      class="_title_ea1127"
    >
      Follow us on social media
    </h2>
    <div
      class="_subtitle_ea1127"
    >
      For the latest product updates and innovations!
    </div>
    <div
      class="_container_ea1127"
    >
      <a
        class="_link_ea1127 _linkX_ea1127"
        href="https://x.com/dyson"
        rel="noreferrer"
        target="_blank"
      >
        <span
          class="_icon_ea1127 _iconX_ea1127"
        />
      </a>
      <a
        class="_link_ea1127"
        href="https://www.instagram.com/dysonusa/"
        rel="noreferrer"
        target="_blank"
      >
        <span
          class="_icon_ea1127 _iconI_ea1127"
        />
      </a>
      <a
        class="_link_ea1127"
        href="https://www.youtube.com/dyson"
        rel="noreferrer"
        target="_blank"
      >
        <span
          class="_icon_ea1127 _iconY_ea1127"
        />
      </a>
      <a
        class="_link_ea1127"
        href="https://www.tiktok.com/@dyson_usa"
        rel="noreferrer"
        target="_blank"
      >
        <span
          class="_icon_ea1127 _iconT_ea1127"
        />
      </a>
    </div>
  </section>
</DocumentFragment>
`;
